### ISSUE TITLE
boost-1.55 compilation issue with gcc-5

### ISSUE THREAD

**Issue by aprokop** [LLNL/spack#2821]
I got a failure compiling boost 1.55.0 with gcc 5.4.0 (got hit by [this](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=197594)). 
```
boost error: no matching function for call to ‘call_once
```
From the thread it seems that 1.56 is fine.

How does one blacklist a specific compiler/compiler version for a specific version of a package?

**Comment by aprokop** [comment]
I guess another solution is to patch the 1.55.0 version. Which one is preferable?

**Comment by alalazo** [comment]
I trust #2822 solves the issue but:
- `boost@1.55.0` release date: November 11th, 2013 19:50 GMT
- `gcc@5.1.0` release date: 2015-04-22

Do we **really** want to play the game of forward porting things using patches when they are outdated and newer versions already exist? Is there a specific need for version 1.55.0 of boost somewhere that cannot be changed?

**Comment by aprokop** [comment]
I agree with you, but not sure if that's avoidable for the project. For the testing, it uses "gold" output, comparing some metrics. Without arguing whether that's the right approach, it really is very dependent on the exact versions of the libraries that it uses so that not to change that output.
I guess, it also depends on what parts of boost it uses, and whether that will affect the output. 

From my perspective, I saw an issue during compilation, and did not spend too much time fixing it. Murphy's law says that somebody will get the same issue some time. Why not fix it?

**Comment by alalazo** [comment]
@aprokop 

> I agree with you, but not sure if that's avoidable for the project. For the testing, it uses "gold" output, comparing some metrics.

Sorry, I don't the context you are referring to: which project? what testing?

> From my perspective, I saw an issue during compilation, and did not spend too much time fixing it. 

The point is: do you need **exactly** `boost@1.55.0` for some reason? If not there are more recent versions of `boost` that you may use and don't have the issue because they were tested by `boost` maintainers on `gcc@5.X.Y`.

`boost` releases [get already tested on many compilers](http://www.boost.org/users/history/version_1_55_0.html), but one thing they can't do is testing them on *future compilers*. Imho we shouldn't start trying to make outdated things with already a replacement work on newer technologies.

> Murphy's law says that somebody will get the same issue some time. Why not fix it?

It also says that you may break 10 things while trying to fix one :smile: I am pretty sure this is not the case here, but again: why adding code to patch `boost@1.55.0` when `boost@1.63.0` compiles just fine and **is tested by boost maintainers**?
